# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/PContent.ipdl
# Commit: baffa8a7b961
# Full Hash: baffa8a7b9618e9a02e4bd808adb71dead99cce5
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2019-11-12 09:43:07
# Regressor Bug: 1589270
# File Overlap Count: 1
# Description:
#   Bug 1589270 - Part 3: Convert nsExternalHelperApp to use BrowsingContext instead of nsIInterfaceRequestor. r=bzbarsky
#   
#   This also converts MaybeCloseWindowHelper, and results in the window close operations being always run in the parent (even without DocumentChannel).
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D49528
# ==============================================================================

diff -r 68bc2ae9b89a -r baffa8a7b961 dom/ipc/PContent.ipdl
--- a/dom/ipc/PContent.ipdl	Fri Nov 08 03:01:03 2019 +0000
+++ b/dom/ipc/PContent.ipdl	Fri Nov 08 04:35:05 2019 +0000
@@ -979,6 +979,24 @@
 
     async PPSMContentDownloader(uint32_t aCertType);
 
+    // Creates a helper for forwarding data from an nsExternalAppHandler
+    // running in the content process, to one running in the parent
+    // process.
+    // Bug 1574372 aims to run nsExternalAppHandler entirely in the
+    // parent so that we can remove this.
+    //
+    // Serializes the uri, loadInfo, contentType, referrer, contentDisposition
+    // headers and contentLength of the channel so that we can make them
+    // available to the parent instance via a nsIChannel helper. Also
+    // passes whether the original channel was an instance of nsIFileChannel.
+    //
+    // aContext is the BrowsingContext that initiated the load, and created the
+    // channel.
+    //
+    // Pass true for aForceSave to always save this content to disk, regardless of
+    // nsIMIMEInfo and other such influences.
+    // Pass true for aShouldCloseWindow to specify that aContext was opened specifically
+    // for this load, and should be closed once we've handled it.
     async PExternalHelperApp(URIParams? uri,
                              LoadInfoArgs? loadInfoArgs,
                              nsCString aMimeContentType,
@@ -989,7 +1007,8 @@
                              int64_t aContentLength,
                              bool aWasFileChannel,
                              URIParams? aReferrer,
-                             nullable PBrowser aBrowser);
+                             BrowsingContext aContext,
+                             bool aShouldCloseWindow);
 
     async PHandlerService();
 